src/main/kotlin/dev/rosewood/roseskyblock/command/argument/IslandWorldGroupArgument.kt

```diff
@@ -111,25 +108,23 @@ class IslandWorldGroupArgument<C> private constructor(
                     )
                 )
 
-            val worldManager = RoseSkyblock.instance.getManager(WorldManager::class)
+            val worldManager = RoseSkyblock.instance.getManager<WorldManager>()
             val islandWorldGroup = worldManager.getIslandWorldGroup(input)
                 ?: return ArgumentParseResult.failure(IslandWorldGroupParseException(input, commandContext))
 
             inputQueue.remove()
             return ArgumentParseResult.success(islandWorldGroup)
         }
 
-        override fun suggestions(commandContext: CommandContext<C>, input: String): List<String> {
-            return RoseSkyblock.instance.getManager(WorldManager::class).worldGroups.map { it.name }
-        }
-
+        // again personal preference, but I like to abuse expression functions like this
+        override fun suggestions(commandContext: CommandContext<C>, input: String)
+            = RoseSkyblock.instance.getManager<WorldManager>().worldGroups.map { it.name }
     }
 
-    class IslandWorldGroupParseException(val input: String, context: CommandContext<*>) : ParserException(
+    class IslandWorldGroupParseException(input: String, context: CommandContext<*>) : ParserException(
```

Comment:
Put val back, just for consistency with cloud command framework. I might change this in the future, but for now I'd rather it mimic an official implementation as much as possible.